x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts

```diff
@@ -356,6 +357,78 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
     return await this.redactSavedObjectNamespaces(savedObject, [options.namespace]);
   }
 
+  public async bulkResolve<T = unknown>(
+    objects: SavedObjectsBulkResolveObject[],
+    options: SavedObjectsBaseOptions = {}
+  ) {
+    try {
+      const args = { objects, options };
+      await this.legacyEnsureAuthorized(
+        this.getUniqueObjectTypes(objects),
+        'bulk_get',
+        options.namespace,
+        { args, auditAction: 'bulk_resolve' }
+      );
+    } catch (error) {
+      objects.forEach(({ type, id }) =>
+        this.auditLogger.log(
+          savedObjectEvent({
+            action: SavedObjectAction.RESOLVE,
+            savedObject: { type, id },
+            error,
+          })
+        )
+      );
+      throw error;
+    }
+
+    const response = await this.baseClient.bulkResolve<T>(objects, options);
+
+    response.resolved_objects.forEach(({ saved_object: { error, type, id } }) => {
+      if (!error) {
+        this.auditLogger.log(
+          savedObjectEvent({
+            action: SavedObjectAction.RESOLVE,
+            savedObject: { type, id },
+          })
+        );
+      }
+    });
+
+    // the generic redactSavedObjectsNamespaces function cannot be used here due to the nested structure of the
```

Comment:
Can we add a comment to `redactSavedObjectsNamespaces` to warn future maintainers that we have a bespoke implementation over here? That way we'll have a better chance of keeping them in sync when it makes sense.

Reply:
Good idea!